src/components/CommentCreateForm/CommentCreateForm.tsx

```diff
@@ -31,19 +31,45 @@ const CommentCreateForm: React.FC<CommentCreateFormProps> = ({ goalId, states, o
     const { user, themeId } = usePageContext();
     const { create } = useCommentResource();
     const [pushState, setPushState] = useState<State | undefined>();
+    const [description, setDescription] = useState<string>();
+    const [focused, setFocused] = useState(false);
+    const [busy, setBusy] = useState(false);
+    const [currentGoal, setCurrentGoal] = useState('');
+    const [prevGoal, setPrevGoal] = useState('');
+
+    if (goalId !== prevGoal) {
+        setCurrentGoal(goalId);
+        setPrevGoal(goalId);
```

Comment:
why do you need prevGoal? can't you compare to currentGoal?

Reply:
https://github.com/taskany-inc/issues/issues/1433